packages/core/src/providers/network/provider.tsx

```diff
@@ -51,7 +52,10 @@ export function NetworkProvider({ children }: NetworkProviderProps) {
 
   const activate = useCallback(
     async (provider: JsonRpcProvider | (EventEmitter & ExternalProvider)) => {
-      const wrappedProvider = provider instanceof JsonRpcProvider ? provider : new Web3Provider(provider)
+      const wrappedProvider =
+        provider instanceof JsonRpcProvider || provider instanceof providers.Web3Provider
```

Comment:
Can it an already existing import from `@ethersproject/providers`? The `ethers` one should be just a re-export

Reply:
It doesn't work when I try that.

Reply:
why?

Reply:
It's not that instance apparently, it gets wrapped by `new Web3Provider(provider)` and then it crashes with weird errors without stacktracke:

```
Error: Method [object Object] not supported.
```

Reply:
looks like there are duplicate dependencies installed 
ethers@5.5.0 vs ethers@5.5.3